Fix/Solved ShiftedDataset issues

pennfranc opened the pull request:
I checked out the TCN example notebook and noticed the performance of the model took a hit after the last big release. I thought it was strange and I believe I found the culprit: When creating the `ShiftedDataset` instance, a `shift` value of 1 was passed instead of `self.output_chunk_length`. Also, in `TorchForecastingModel._produce_prediction` the indexing had to be updated to reflect the new data structures. The latter bug only affected the `ShiftedDataset`, which is probably why it went unnoticed. With these changes the performance on the examples in the TCN notebook dramatically improved. Please let me know if this makes sense.
If there is any way this could be released soon, that would be very cool. Since without this the TCNModel is a bit useless, and I feel responsible for it haha

hrzn reviewed (approved):
Good catch! Thanks a lot @pennfranc. Let's include that in the next release (which I want to make soon). Before we merge:
* Could you move the length check to __getitem__() method, and
* Are there some unit tests accuracy lower bounds that we could make tighter, to ensure that we properly test the accuracy of TCN?

pennfranc commented on darts/utils/data/shifted_dataset.py:
```diff
@@ -66,6 +66,9 @@ def __init__(self,
                      'The provided sequence of target series must have the same length as '
                      'the provided sequence of covariate series.')
 
+        raise_if_not(min(len(ts) for ts in self.target_series) - length - shift + 1 > 0,
```
Makes sense, done!

pennfranc commented:
@hrzn I added a simple performance test on a dummy time series (sine + linear). I also moved the length check like you suggested. If you think it looks good feel free to merge!

hrzn commented:
Awesome, thanks @pennfranc !
We have some issues with some older checks still showing up I don't know why... but all the required ones are green, so I'll already merge it.
